src/microsAuthentication/content.md

```diff
@@ -0,0 +1,1159 @@
+	if err != nil {
+		panic(err)
+	}
+	serviceToken = string(b)
+}
+```
+
+Then, the service token is passed on to the call to the Secret store service in a HTTP header, X-Client-Id:
+
+```go
+func handleIndex(w http.ResponseWriter, r *http.Request) {
+
+	...
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", serviceConnstring, nil)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Add("X-Client-Id", serviceToken)
+	resp, err := client.Do(req)
+	..
+```
+
+The following YAML manifest is used to deploy the API service:
+
+```yaml|highlight=27-27|title=deployment.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: api
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: api
+  namespace: api
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  namespace: api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: api
+  template:
+    metadata:
+      labels:
+        app: api
+    spec:
+      serviceAccount: api
+      containers:
+      - name: app
+        image: amitsaha/k8s-sa-volume-demo-api:sa-1
+        env:
+        - name: LISTEN_ADDR
+          value: ":8080"
+        - name: SECRET_STORE_CONNSTRING
+          value: "http://app.secret-store.svc.cluster.local"
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+  namespace: api
+spec:
+  type: NodePort
+  selector:
+    app: api
+  ports:
+    - port: 8080
+      targetPort: 8080
+```
+
+You will notice that there is nothing special about the deployment manifest above other than specifying a service account to the deployment.
+
+Let's move onto the secret store service. You can find the complete application in `service_accounts/secret-store/main.go`.
+
+The secret store service does two key things:
+
+1. It retrieves the value of the X-Client-Id header from the incoming request.
+1. It then invokes the Kubernetes TokenReview API with the value retrieved to check if it is a valid token or not.
+
+Step (1) is performed by the following code:
+
+```go
+clientId := r.Header.Get("X-Client-Id")
+	if len(clientId) == 0 {
+		http.Error(w, "X-Client-Id not supplied", http.StatusUnauthorized)
+		return
+	}
+```
+
+Then, step (2) is performed using the Kubernetes Go client.
+
+First, we create a `ClientSet` object:
+
+```go
+config, err := rest.InClusterConfig()
+clientset, err := kubernetes.NewForConfig(config)
+```
+
+The `InClusterConfig()` function automatically reads the service account token for the pod and hence you do not have to manually specify the filesystem path.
+
+Then, you construct a `TokenReview` object specifying the token you want to validate in the `Token` field:
+
+```go
+tr := authv1.TokenReview{
+	Spec: authv1.TokenReviewSpec{
+	    Token: clientId,
+	},
+}
+```
+
+Then, create a `TokenReview` request:
+
+```go
+result, err := clientset.AuthenticationV1().TokenReviews().Create(ctx, &tr, metav1.CreateOptions{})
+```
```

Comment:
You broke the code line-by-line.
The idea is good, but I think it could be refactored to provide more context.
Keep the line by line, but should the snippet truncated in full.
Then use `highlight` in the code snippet to highlight specific parts.
In that way you offer context, but you can explain individual parts.